awx/main/dispatch/worker/callback.py

```diff
@@ -26,7 +30,37 @@ class CallbackBrokerWorker(BaseWorker):
 
     MAX_RETRIES = 2
 
+    def __init__(self):
+        self.buff = {}
+        self.last_write = tz_now()
+
+    def read(self, queue):
+        try:
+            return queue.get(block=True, timeout=.25)
+        except QueueEmpty:
+            return 'FLUSH'
+
+    def flush(self):
+        should_flush = (
+            any([len(events) == 2000 for events in self.buff.values()]) or
+            (tz_now() - self.last_write).seconds > .1
+        )
+        if should_flush:
+            now = tz_now()
+            for cls, events in self.buff.items():
+                for e in events:
+                    e.created = e.modified = now
+                cls.objects.bulk_create(events)
```

Comment:
Could we add some diagnostic for assess how many events are created in a single bulk create? I would even consider creating a UUID for each group with more than 1 event, so that someone could run complex analysis using the available data after the fact.

But without going that far, event type & number in a log statement would suffice.

Can you also confirm - this list of events may contain events from separate jobs, correct? We may want to make a testing note to cover _multiple_ high output jobs running simultaneously.

Reply:
I like the idea of a `.debug` level log line for this.

As for simultaneous jobs, that *is* something I've been testing from the beginning - I've been running this on a beefy ec2 vm with slices > 10, and events go where they belong (all three jobs have an equal number of events).